ingester/src/catalog_update.rs

```diff
@@ -0,0 +1,131 @@
+//! Module to update the catalog during ingesting
+
+use std::sync::Arc;
+
+use iox_catalog::interface::Tombstone;
+use snafu::{ResultExt, Snafu};
+
+use crate::{compact::IoxMetadata, data::PersistingBatch, handler::IngestHandlerImpl};
+
+#[derive(Debug, Snafu)]
+#[allow(missing_copy_implementations, missing_docs)]
+pub enum Error {
+    #[snafu(display(
+        "The Object Store ID of the Persisting Batch and its metadata are not matched"
+    ))]
+    NoMatch,
```

Comment:
This error variant could include the ID of the persisting batch and the ID of the metadata to help with debugging, something like:

```suggestion
    #[snafu(display(
        "The Object Store ID of the Persisting Batch ({}) and its metadata ({}) are not matched",
        persisting_batch_id,
        metadata_id,
    ))]
    NoMatch {
        persisting_batch_id: Uuid,
        metadata_id: Uuid,
    },
```